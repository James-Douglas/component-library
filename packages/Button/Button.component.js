import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ManorProvider } from '@comparethemarketau/manor-provider';
import { useId } from '@comparethemarketau/manor-hooks';
import {
  StyledIcon,
  BaseTag,
  StyledButtonWrap,
} from './Button.styles';

const Button = ({
  id: propsId,
  variant,
  inverted,
  disabled,
  icon,
  iconAlign,
  href,
  target,
  rel,
  handleClick,
  tagType,
  width,
  style,
  children,
  className,
  theme,
}) => {
  const id = useId(propsId);
  const renderContent = () => {
    if (icon) {
      return (
        <>
          <StyledIcon>
            <FontAwesomeIcon icon={icon} size="1x" />
          </StyledIcon>
          {children}
        </>
      );
    }
    return <>{children}</>;
  };
  const renderButton = () => {
    const isButton = !href.length > 0;
    return (
      <BaseTag
        as={isButton ? 'button' : 'a'}
        onClick={handleClick}
        id={id}
        variant={variant}
        iconAlign={iconAlign}
        inverted={inverted}
        disabled={disabled}
        href={isButton ? null : href}
        target={isButton ? null : target}
        rel={isButton ? null : rel}
        type={isButton ? tagType : null}
        style={style}
        className={className}
      >
        {renderContent()}
      </BaseTag>
    );
  };

  return (
    <ManorProvider theme={theme}>
      <StyledButtonWrap variant={variant} width={width}>
        {renderButton()}
      </StyledButtonWrap>
    </ManorProvider>
  );
};

Button.propTypes = {
  /**
   * Unique identifier for the button
   */
  id: PropTypes.string,
  /**
   * The button/link contents
   */
  children: PropTypes.string,
  /**
   * Defines the type of button to be used. Defaults to `primary`. available types are `primary`, `secondary`,
   * `tertiary`, `text`, `link`, and `footer-link`.
   */
  variant: PropTypes.oneOf(['hero', 'primary', 'secondary', 'tertiary']),
  /**
   * Inverts the button. Tertiary doesn't have an inverted mode as its a ghost button
   */
  inverted: PropTypes.bool,
  /**
   * Defines if the button is disabled or not. Defaults to false.
   */
  disabled: PropTypes.bool,
  /**
   * The icon to be supplied with the button. Defaults to none.
   */
  icon: PropTypes.oneOfType([
    PropTypes.shape({
      prefix: PropTypes.string,
      iconName: PropTypes.string,
      icon: PropTypes.array, // eslint-disable-line
    }),
    PropTypes.string,
  ]),
  /**
   * Shift the icon to the right, content to the left. Defaults to false.
   */
  iconAlign: PropTypes.oneOf(['left', 'right']),
  /**
   * Defines the `<a>` link. Defaults to `#`.
   */
  href: PropTypes.string,
  /**
   * Defines if the `<a>` should have a target attribute.
   */
  target: PropTypes.string,
  /**
   * Specifies the rel, for example "nofollow" can be supplied
   */
  rel: PropTypes.string,
  /**
   * Specifies the type attribute to add to a button tag
   */
  tagType: PropTypes.string,
  /**
   * Manually set the width of a button
   */
  width: PropTypes.string,
  /**
   * Overide the css with inline styles if needed
   */
  style: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ])),
  /**
   * Classes to be applied to the Button component
   */
  className: PropTypes.string,
  /**
   * Pass the button a custom click function
   */
  handleClick: PropTypes.func,
  // eslint-disable-next-line react/forbid-prop-types
  /**
   * Manor theme, if not provided the ctm theme will be used.
   */
  // eslint-disable-next-line react/forbid-prop-types
  theme: PropTypes.object,
};

Button.defaultProps = {
  id: null,
  variant: 'hero',
  inverted: false,
  children: '',
  disabled: false,
  icon: null,
  iconAlign: 'left',
  href: '',
  target: '#',
  rel: '',
  tagType: '',
  style: {},
  width: '100%',
  handleClick: null,
  className: '',
  theme: undefined,
};

export default Button;
