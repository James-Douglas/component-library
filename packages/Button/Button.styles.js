import styled, { css } from 'styled-components';

export const StyledIcon = styled.div`
  pointer-events: none;
  display: inherit;
`;

const disabledStyles = css`
  cursor: not-allowed;
  background: ${({ theme }) => theme.button.disabledBackgound};
  color: ${({ theme }) => theme.button.disabledText};
  box-shadow: ${({ theme }) => theme.button.shadows.disabled};
  border: none;
  :hover {
    box-shadow: ${({ theme }) => theme.button.shadows.disabled};
    background: ${({ theme }) => theme.button.disabledBackgound};
  }
`;

const baseButtonStyles = css`
  font-family: ${({ theme }) => theme.fontFamily};
  border: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  text-decoration: none;
  width: 100%;
  font-size: ${({ theme }) => theme.fontSize.base};
  font-style: ${({ theme }) => theme.fontWeight.normal};
  line-height: ${({ theme }) => theme.lineHeight.tighter};
  box-shadow: ${({ theme }) => theme.button.shadows.default};
  border-radius: ${({ theme }) => theme.button.borderRadius};
  padding: ${({ theme }) => theme.spacing[16]};
  margin-bottom: ${({ theme }) => theme.spacing[16]};
  transition: ${({ theme }) => theme.transition.default};
  font-weight: ${({ theme }) => theme.button.fontWeight};
  > ${StyledIcon} {
    margin-right: ${({ theme, iconButton }) => (iconButton ? 0 : theme.spacing[8])};
    ${({ theme, iconButton }) => iconButton && css`
    justify-content: center;
    min-width: ${theme.spacing[24]};
    `};
  }
  ${({ iconAlign, theme }) => iconAlign === 'right' && css`
    flex-direction: row-reverse;
    > ${StyledIcon} {
      margin-right: 0;
      margin-left: ${theme.spacing[8]};
    }
  `}
  :focus {
    outline: none;
  }
  :hover {
    box-shadow: ${({ theme }) => theme.button.shadows.hover};
  }
  :active, :focus {
    box-shadow: ${({ theme }) => theme.button.shadows.focused};
  }
`;

const heroVariant = css`
  color: ${({ theme }) => theme.button.hero.color};
  fill: ${({ theme }) => theme.button.hero.fill};
  background: ${({ theme }) => theme.button.hero.background};
  :hover {
    background: ${({ theme }) => theme.button.hero.backgroundHover};
  }
  ${({ inverted }) => inverted && css`
    background: ${({ theme }) => theme.button.hero.backgroundInverted};
    color: ${({ theme }) => theme.button.hero.colorInverted};
    fill: ${({ theme }) => theme.button.hero.fillInverted};
    box-shadow: ${({ theme }) => theme.button.shadows.default};
    :hover {
      background: ${({ theme }) => theme.button.hero.backgroundInvertedHover};
    }
  `}
`;

const primaryVariant = css`
  color: ${({ theme }) => theme.button.primary.color};
  fill: ${({ theme }) => theme.button.primary.fill};
  background: ${({ theme }) => theme.button.primary.background};
  :hover {
    background: ${({ theme }) => theme.button.primary.backgroundHover};
  }
  ${({ inverted }) => inverted && css`
    background: ${({ theme }) => theme.button.primary.backgroundInverted};
    color: ${({ theme }) => theme.button.primary.colorInverted};
    fill: ${({ theme }) => theme.button.primary.fillInverted};
    :hover {
      background: ${({ theme }) => theme.button.primary.backgroundInvertedHover};
    }
  `}
`;

const secondaryVariant = css`
  color: ${({ theme }) => theme.button.secondary.color};
  fill: ${({ theme }) => theme.button.secondary.fill};
  border: ${({ theme }) => theme.button.secondary.border};
  background: ${({ theme }) => theme.button.secondary.background};
  :hover {
    background: ${({ theme }) => theme.button.secondary.backgroundHover};
  }
  ${({ inverted }) => inverted && css`
    background: ${({ theme }) => theme.button.secondary.backgroundInverted};
    color: ${({ theme }) => theme.button.secondary.colorInverted};
    border: ${({ theme }) => theme.button.secondary.borderInverted};
    fill: ${({ theme }) => theme.button.secondary.fillInverted};
    :hover {
      background: ${({ theme }) => theme.button.secondary.backgroundInvertedHover};
    }
  `}
`;

const tertiaryVariant = css`
  box-shadow: none;
  padding: ${({ theme }) => theme.spacing[8]};
  color: ${({ theme }) => theme.button.tertiary.color};
  fill: ${({ theme }) => theme.button.tertiary.fill};
  background: ${({ theme }) => theme.button.tertiary.background};
  :hover {
    background: ${({ theme }) => theme.button.tertiary.backgroundHover};
  }
  :focus {
    color: ${({ theme }) => theme.button.tertiary.colorhover};
    fill: ${({ theme }) => theme.button.tertiary.colorhover};
  }
`;

export const BaseTag = styled.div`
  ${baseButtonStyles}
  ${({ variant }) => {
    switch (variant) {
      case 'hero':
        return heroVariant;
      case 'primary':
        return primaryVariant;
      case 'secondary':
        return secondaryVariant;
      case 'tertiary':
        return tertiaryVariant;
      default:
        return heroVariant;
    }
  }}
  ${({ disabled, href }) => (disabled && !href) && disabledStyles};
`;

export const StyledButtonWrap = styled.span`
  display: flex;
  flex-wrap: wrap;
  flex: 0 1 100%;
  width: ${({ width }) => width};
  & {
    font-family: ${({ theme }) => theme.fontFamily};
  }
`;
