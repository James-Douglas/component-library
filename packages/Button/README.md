# Button

## Installation

`yarn add @comparethemarketau/manor-button`

## Documentation

For technical documentation, see [Storybook](https://services.dev.comparethemarket.cloud/manor/?path=/docs/components-button--hero-button).
Design system documentation can be found here [Bento](https://zeroheight.com/9942937b5/p/437255-buttons/b/97d6bd)
