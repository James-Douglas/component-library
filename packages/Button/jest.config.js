const baseConfig = require('../../jest.config.base');

module.exports = {
  ...baseConfig,
  roots: [
    '<rootDir>/packages/Button',
  ],
  collectCoverageFrom: [
    '*.{js,jsx,ts,tsx}',
  ],
  name: 'Button',
  displayName: 'Button',
  rootDir: '../..',
};
