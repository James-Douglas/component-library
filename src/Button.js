import React from 'react'
import "./index.css"

export default function Button() {
  return (
    <div className="btn text-invalid w-256">
      button
    </div>
  )
}
