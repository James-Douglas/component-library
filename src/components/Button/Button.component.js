import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon/Icon.component';
import styles from './styles';

const Button = ({
  id, type, onDark, size, content, disabled, icon, iconSize, iconAlignRight, href, target, rel, handleClick,
}) => {
  const isInlineBlock = href && type !== 'footer-link' ? 'inline-block' : '';

  const renderContent = () => {
    if (icon) {
      return (
        <>
          <style jsx>{styles}</style>
          <div className="btn-icon">
            <Icon name={icon} iconSize={iconSize} />
          </div>
          {content}
        </>
      );
    }
    return <>{content}</>;
  };

  const renderButton = () => {
    const isButton = type === 'primary' || type === 'secondary';
    const Tag = isButton ? 'button' : 'a';

    return (
      <>
        <style jsx>{styles}</style>
        <Tag
          onClick={handleClick}
          id={id}
          className={`
            manor-rich-text 
            ${size}
            ${type}
            ${isButton ? 'manor-button' : 'manor-button-link'} 
            ${onDark ? 'onDark' : ''}
            ${content === '' ? 'center' : ''}
            ${iconAlignRight ? 'align-right' : ''}
          `}
          disabled={disabled}
          href={isButton ? null : href}
          target={isButton ? null : target}
          rel={isButton ? null : rel}
        >
          {renderContent()}
        </Tag>
      </>
    );
  };

  return (
    <>
      <style jsx>{styles}</style>
      <span className={`manor-button-wrap ${isInlineBlock}`}>
        {renderButton()}
      </span>
    </>
  );
};

Button.propTypes = {
  id: PropTypes.string.isRequired,
  type: PropTypes.string,
  onDark: PropTypes.bool,
  size: PropTypes.string,
  content: PropTypes.string,
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  iconSize: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  iconAlignRight: PropTypes.bool,
  href: PropTypes.string,
  target: PropTypes.string,
  rel: PropTypes.string,
  handleClick: PropTypes.func,
};

Button.defaultProps = {
  type: 'primary',
  onDark: false,
  size: 'md',
  content: '',
  disabled: false,
  icon: '',
  iconSize: 2,
  iconAlignRight: false,
  href: '',
  target: '#',
  rel: '',
  handleClick: () => {},
};

export default Button;
