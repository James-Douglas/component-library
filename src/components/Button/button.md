# Button component<br/>
## Usage
~~~js
<Button
	{id}
	{disabled}
	{btnType}
	{btnMode}
	{btnSize}
	{href}
	{target}
	{icon}
	{size}
	{content}
	on:click
/>
~~~

## Props
The button can be styled with an icon, configurable for design approved modes. These are: `primary`, `Secondary`, `text`, `link` and `footer-link`. `secondary`, `text` and `link` also have an `onDark` mode for dark backgrounds.

When `text`, `link` and `footer-link` and supplied as a `btnType`, the html structure with be rendered as an `<a>` element, with removed padding/button styling.

1. `id` *(required) string*
    * The button ID.
2. `disabled` *(optional) boolean*
	  * Defines if the button is disabled or not. Defaults to false.
3. `btnType` *(optional) string*
	  * Defines the type of button to be used. Defaults to `primary`. Other types are `secondary`, `text`, `link` and `footer-link`.
4. `btnMode` *(optional) string*
    * Defines the mode to be in. Can be changed to `onDark` which changes the buttons for `secondary`, `text`, `link` and `footer-link`. `primary` does not have a `onDark` mode.
5. `btnSize` *(optional) string*
	  * Defines the size of button to be used, `sm`, `md` and `lg`. Defaults to `md`.
6. `href` *(optional) string*
	  * Defines the `<a>` a link. Defaults to `javascript:void(0);`.
7. `target` *(optional) string*
	  * Defines if the `<a>` should have a target attribute. 
8. `icon` *(optional) string*
    * The icon to be supplied with the button. Defaults to none.
9. `size` *(optional) string*
    * The icon sizing. Defaults to '2'
10. `content` *(required) string*
    * The button text. Can render html if required.

## Events
A click event is forwarded from the component to be used by its consumer if needed. For example, on usage you can bind a function to `on:click` which will return the event, useful to pull out its ID etc.
