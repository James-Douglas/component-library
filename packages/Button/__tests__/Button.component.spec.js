import React from 'react';
import 'jest-styled-components';
import { faInfoCircle } from '@fortawesome/pro-regular-svg-icons/faInfoCircle';
import { render } from '@testing-library/react';
import { ctmTheme } from '@comparethemarketau/manor-themes';
import Button from '../Button.component';

describe('Button', () => {
  it('renders correctly with minimal props', () => {
    const { container } = render(<Button id="test-id" />);
    expect(container).toMatchSnapshot();
  });

  it('renders with props', () => {
    const { container, getByText } = render(
      <Button id="test-id" variant="hero" icon={faInfoCircle} iconAlign="right">
        test content
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(getByText('test content')).toBeDefined();
    expect(btn).toHaveStyleRule('background', ctmTheme.button.hero.background);
    expect(btn).toHaveStyleRule('flex-direction', 'row-reverse');
    expect(btn).toHaveStyleRule('box-shadow', ctmTheme.button.shadows.default);
    expect(container).toMatchSnapshot();
  });

  it('renders a hero button', () => {
    const { container } = render(
      <Button id="test-id" variant="hero">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('BUTTON');
    expect(btn).not.toHaveStyle('flex-direction: row-reverse');
    expect(btn).toHaveStyleRule('background', ctmTheme.button.hero.background);
  });

  it('renders an inverted hero button', () => {
    const { container } = render(
      <Button id="test-id" variant="hero" inverted>
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('BUTTON');
    expect(btn).not.toHaveStyle('flex-direction: row-reverse');
    expect(btn).toHaveStyleRule('background', ctmTheme.button.hero.backgroundInverted);
  });

  it('renders a hero button as a link when the props are supplied', () => {
    const { container } = render(
      <Button id="test-id" variant="hero" href="#">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('A');
    expect(btn).toHaveStyleRule('background', ctmTheme.button.hero.background);
  });

  it('renders a primary button', () => {
    const { container } = render(
      <Button id="test-id" variant="primary">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('BUTTON');
    expect(btn).toHaveStyleRule('background', ctmTheme.button.primary.background);
    expect(btn).toHaveStyleRule('color', ctmTheme.button.primary.color);
  });

  it('renders an inverted primary', () => {
    const { container } = render(
      <Button id="test-id" variant="primary" inverted>
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('BUTTON');
    expect(btn).toHaveStyleRule('background', ctmTheme.button.primary.backgroundInverted);
    expect(btn).toHaveStyleRule('color', ctmTheme.button.primary.colorInverted);
  });

  it('renders a secondary button', () => {
    const { container } = render(
      <Button id="test-id" variant="secondary">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('BUTTON');
    expect(btn).toHaveStyleRule('background', ctmTheme.button.secondary.background);
    expect(btn).toHaveStyleRule('border', ctmTheme.button.secondary.border);
  });

  it('renders a secondary button as a link when the props are supplied', () => {
    const { container } = render(
      <Button id="test-id" variant="secondary" href="#">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn).toHaveStyleRule('background', ctmTheme.button.secondary.background);
    expect(btn.nodeName).toBe('A');
    expect(btn).toHaveStyleRule('border', ctmTheme.button.secondary.border);
  });

  it('renders a secondary button in mode = inverted, when the props are supplied', () => {
    const { container } = render(
      <Button id="test-id" variant="secondary" inverted>
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn).toHaveStyleRule('border', ctmTheme.button.secondary.borderInverted);
    expect(btn).toHaveStyleRule('box-shadow', ctmTheme.button.shadows.default);
  });

  it('renders a secondary button in mode = inverted as a link, when the props are supplied', () => {
    const { container } = render(
      <Button id="test-id" variant="secondary" inverted href="#">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn).toHaveStyleRule('border', ctmTheme.button.secondary.borderInverted);
    expect(btn.nodeName).toBe('A');
    expect(btn).toHaveStyleRule('box-shadow', ctmTheme.button.shadows.default);
  });

  it('renders a tertiary button', () => {
    const { container } = render(
      <Button id="test-id" variant="tertiary">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('BUTTON');
    expect(btn).toHaveStyleRule('box-shadow', ctmTheme.button.shadows.disabled);
  });

  it('renders a tertiary button as a link when the props are supplied', () => {
    const { container } = render(
      <Button id="test-id" variant="tertiary" href="#">
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn.nodeName).toBe('A');
    expect(btn).toHaveStyleRule('box-shadow', ctmTheme.button.shadows.disabled);
  });

  it('renders a disabled button', () => {
    const { container } = render(
      <Button id="test-id" variant="hero" disabled>
        test thing
      </Button>,
    );
    const btn = container.querySelector('#test-id');
    expect(btn).toHaveStyleRule('cursor', 'not-allowed');
  });
});
