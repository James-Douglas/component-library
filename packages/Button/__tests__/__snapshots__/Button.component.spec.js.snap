// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with minimal props 1`] = `
.c0 {
  position: fixed;
  top: 7.6rem;
  right: 0;
  overflow: hidden;
  z-index: 2147483647;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: Capitalize;
  width: 100%;
  font-size: 1.6rem;
  font-style: 400;
  line-height: 1.2;
  box-shadow: 0px 0.3rem 0.8rem rgba(0,0,0,0.2);
  border-radius: 0.4rem;
  padding: 1.6rem;
  margin-bottom: 1.6rem;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  font-weight: 700;
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #0f8a28;
}

.c3 > .sc-AxiKw {
  margin-right: 0.8rem;
}

.c3:focus {
  outline: none;
}

.c3:hover {
  box-shadow: 0px 0.5rem 0.8rem rgba(0,0,0,0.2);
}

.c3:active,
.c3:focus {
  box-shadow: 0px 0.3rem 0.8rem rgba(0,0,0,0.1);
}

.c3:hover {
  background: #0d7522;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  width: 100%;
}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
  <span
    class="c2"
    width="100%"
  >
    <button
      class="c3"
      id="test-id"
      type=""
    >
      
    </button>
  </span>
</div>
`;

exports[`Button renders with props 1`] = `
.c0 {
  position: fixed;
  top: 7.6rem;
  right: 0;
  overflow: hidden;
  z-index: 2147483647;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  pointer-events: none;
  display: inherit;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: Capitalize;
  width: 100%;
  font-size: 1.6rem;
  font-style: 400;
  line-height: 1.2;
  box-shadow: 0px 0.3rem 0.8rem rgba(0,0,0,0.2);
  border-radius: 0.4rem;
  padding: 1.6rem;
  margin-bottom: 1.6rem;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  font-weight: 700;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #0f8a28;
}

.c3 > .c4 {
  margin-right: 0.8rem;
}

.c3 > .c4 {
  margin-right: 0;
  margin-left: 0.8rem;
}

.c3:focus {
  outline: none;
}

.c3:hover {
  box-shadow: 0px 0.5rem 0.8rem rgba(0,0,0,0.2);
}

.c3:active,
.c3:focus {
  box-shadow: 0px 0.3rem 0.8rem rgba(0,0,0,0.1);
}

.c3:hover {
  background: #0d7522;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  width: 100%;
}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
  <span
    class="c2"
    width="100%"
  >
    <button
      class="c3"
      id="test-id"
      type=""
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-info-circle fa-w-16 fa-1x "
          data-icon="info-circle"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm0-338c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
          />
        </svg>
      </div>
      test content
    </button>
  </span>
</div>
`;
